### PR TITLE
Add object upload operator

### PR DESCRIPTION
**Pull request by arielshaqed**
Tested manually on a standalone Airflow, after removing the commit_sensor step (which is not relevant to file uploads, but OTOH doesn't work with `airflow standalone` because of the required parallelism...).

### CODE REVIEW

**Comment by arielshaqed** [comment]
#18 to make the tests pass.  It's based on this one, so it must be good :-)

File: lakefs_provider/hooks/lakefs_hook.py [hunk 1]
```diff
@@ -57,6 +57,16 @@ def commit(self, repo: str, branch: str, msg: str, metadata: Dict[str, Any] = No
...
 
+    def upload(self, repo: str, branch: str, path: str, content: IO) -> str:
+        client = self.get_conn()
+        upload = client.objects.upload_object(
+            repository=repo,
+            branch=branch,
+            path=path,
+            content=content)
+
+        return upload
```
Comment (lynnro314) [inline lakefs_provider/hooks/lakefs_hook.py:68]: Not sure - is the return type in this case is a string?

File: lakefs_provider/operators/upload_operator.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    def __init__(self, lakefs_conn_id: str, repo: str, branch: str, path: str, content: IO = None, **kwargs: Any) -> None:
+        super().__init__(**kwargs)
+        self.lakefs_conn_id = lakefs_conn_id
+        self.repo = repo
+        self.branch = branch
+        self.path = path
+        self.content = content
+        self.task_id = kwargs.get("task_id")
+
+    def execute(self, context: Dict[str, Any]) -> Any:
```
Comment (lynnro314) [inline lakefs_provider/operators/upload_operator.py:44]: I saw that this is similar to all lakeFS operators, but why this is the type of context?

**Review by lynnro314 — APPROVED** [review]
Minor comments, but LGTM :)

**Comment by arielshaqed** [comment]
Thanks!

#18 will allow tests to pass (and will now be auto-transformed into a merge-to-main).

**Pull request merged**